### PR TITLE
Add support for assets-erc20 precompile

### PR DESCRIPTION
**Pull request by Agusrodri**
This PR allows to manage foreign assets through the assets-erc20 precompile. It also adds typescript tests related to the overall functionality.

- [x] TODO: update branches once https://github.com/Moonsong-Labs/moonkit/pull/27 is merged

### CODE REVIEW

File: container-chains/templates/frontier/runtime/src/xcm_config.rs [hunk 1]
```diff
@@ -288,7 +289,34 @@ impl pallet_asset_rate::AssetKindFactory<AssetId> for ForeignAssetBenchmarkHelpe
     }
 }
 
-pub type AssetId = u16;
+// Instruct how to go from an H160 to an AssetID
+// We just take the lowest 16 bits
+impl AccountIdAssetIdConversion<AccountId, AssetId> for Runtime {
+    /// The way to convert an account to assetId is by ensuring that the prefix is 0XFFFFFFFF
```
Comment (girazoki) [inline container-chains/templates/frontier/runtime/src/xcm_config.rs:295]:
```suggestion
    /// The way to convert an account to assetId is by ensuring that the prefix is [0xFF, 18]
```

File: container-chains/templates/frontier/runtime/src/xcm_config.rs [hunk 2]
```diff
@@ -288,7 +289,34 @@ impl pallet_asset_rate::AssetKindFactory<AssetId> for ForeignAssetBenchmarkHelpe
...
+    /// and by taking the lowest 16 bits as the assetId
```
Comment (girazoki) [inline container-chains/templates/frontier/runtime/src/xcm_config.rs:296]:
```suggestion
    /// and by taking the lowest 2 bytes as the assetId
```

File: container-chains/templates/frontier/runtime/src/xcm_config.rs [hunk 3]
```diff
@@ -288,7 +289,34 @@ impl pallet_asset_rate::AssetKindFactory<AssetId> for ForeignAssetBenchmarkHelpe
...
+    // The opposite conversion
+    fn asset_id_to_account(prefix: &[u8], asset_id: AssetId) -> AccountId {
+        let mut data = [0u8; 20];
+        data[0..18].copy_from_slice(prefix);
+        data[18..20].copy_from_slice(&asset_id.to_be_bytes());
+        AccountId::from(data)
+    }
+}
+
+pub type AssetId = u128;
```
Comment (girazoki) [inline container-chains/templates/frontier/runtime/src/xcm_config.rs:319]:
```suggestion
```

File: container-chains/templates/frontier/runtime/src/xcm_config.rs [hunk 4]
```diff
@@ -288,7 +289,34 @@ impl pallet_asset_rate::AssetKindFactory<AssetId> for ForeignAssetBenchmarkHelpe
     }
 }
 
-pub type AssetId = u16;
```
Comment (girazoki) [inline container-chains/templates/frontier/runtime/src/xcm_config.rs:291]: This should not be removed

Comment (Agusrodri) [reply container-chains/templates/frontier/runtime/src/xcm_config.rs:291]: Yes this should be set to `u16`. I changed it for testing purposes to `u128` and I forgot to put it back again to its original value.

**Review by girazoki — APPROVED** [review]
looks good to me!

**Review by tmpolaczyk — APPROVED** [review]

**Pull request merged**